benchmarks/mmmu/mmmu.py

```diff
@@ -0,0 +1,169 @@
+            )
+    return content_list
+
+
+MULT_CHOICE_PROMPT = r"""
+Answer with the option's letter from the given choices directly.
+{prompt}
+""".strip()
+
+
+def mmmu_task_multiple_choice(dataset):
+    multiple_choice_questions_hf = dataset.filter(
+        lambda example: example["question_type"] == "multiple-choice"
+    )
+    multiple_choice_questions_inspect = MemoryDataset(
+        samples=[
+            record_to_sample_multiple_choice(data)
+            for data in multiple_choice_questions_hf
+        ],
+        name="MMMU-Multiple-Choice",
+        location="MMMU/MMMU",
+    )
+
+    return Task(
+        dataset=multiple_choice_questions_inspect,
+        plan=[prompt_template(MULT_CHOICE_PROMPT), multiple_choice()],
+        scorer=choice(),
+        config=GenerateConfig(temperature=0),
+    )
+
+
+OPEN_PROMPT = r"""
+Answer the question using a single word or phrase.
+{prompt}
+""".strip()
+
+
+def mmmu_task_open(dataset):
+    open_ended_questions_hf = dataset.filter(
+        lambda example: example["question_type"] == "open"
+    )
+    open_ended_questions_inspect = MemoryDataset(
+        samples=[record_to_sample_open_ended(data) for data in open_ended_questions_hf],
+        name="MMMU-Open-Ended",
+        location="MMMU/MMMU",
+    )
+
+    return Task(
+        dataset=open_ended_questions_inspect,
+        plan=[prompt_template(OPEN_PROMPT), generate()],
+        scorer=model_graded_fact(),
+        config=GenerateConfig(temperature=0),
+    )
+
+
+subjects_list = [
+    "Accounting",
+    "Agriculture",
+    "Architecture_and_Engineering",
+    "Art",
+    "Art_Theory",
+    "Basic_Medical_Science",
+    "Biology",
+    "Chemistry",
+    "Clinical_Medicine",
+    "Computer_Science",
+    "Design",
+    "Diagnostics_and_Laboratory_Medicine",
+    "Economics",
+    "Electronics",
+    "Energy_and_Power",
+    "Finance",
+    "Geography",
+    "History",
+    "Literature",
+    "Manage",
+    "Marketing",
+    "Materials",
+    "Math",
+    "Mechanical_Engineering",
+    "Music",
+    "Pharmacy",
+    "Physics",
+    "Psychology",
+    "Public_Health",
+    "Sociology",
+]
+
+
+# Allow multiple-choice/open-ended tasks to be run separately
+@task
+def mmmu_multiple_choice():
+    path = "MMMU/MMMU"
+    datasets = [load_dataset(path, subject, split="dev") for subject in subjects_list]
+    combined_dataset = concatenate_datasets(datasets)
+
+    return mmmu_task_multiple_choice(combined_dataset)
+
+
+# Allow multiple-choice/open-ended tasks to be run separately
+@task
+def mmmu_open():
+    path = "MMMU/MMMU"
+    datasets = [load_dataset(path, subject, split="dev") for subject in subjects_list]
+    combined_dataset = concatenate_datasets(datasets)
+
+    return mmmu_task_open(combined_dataset)
+
+
+# Map records to Inspect Sample for multiple-choice questions
+def record_to_sample_multiple_choice(record):
+    content_list = create_content_list(record)
+    answers_list = ast.literal_eval(record["options"])
```

Comment:
Why do you need to `ast.literal_eval` here? How does `record["options"]` look like? Is this a JSON string, you might want to use json instead to parse it.

Reply:
`record["options"]` is a string, of a list of strings (see below), so I opted to use `ast.literal_eval` to parse this.
![image](https://github.com/user-attachments/assets/c62eb0b8-6744-4b7e-8676-f29562b87055)